include/mppic/models/sampling_std.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+namespace mppi::models
+{
+
+struct SamplingStd
+{
```

Comment:
Isn't this still a control constraint? Wouldn't making this struct in the `control_constraints.hpp` file still make sense? I don't like having a bunch of these small files with just struct defs if we can help it.

Also, all these new files needs the copyright line for linters

Reply:
Or even put the stds inside of the constraints struct, since that is part of the constraint

Reply:
This struct consist of parameters for sampling strategy (std of gaussian distribution).  Semantically very different for me, but i could be wrong. We could have in future different sampling strategies, so why not separate concerns ?

Reply:
Even if there are future "constraints" of various forms, I think they're still constraints to belong in that struct. I think it would make sense to group them together

Reply:
Constraints impose limits on state vars. This gaussian standard deviation set up sampling strategy params / optimization problem search params. std does not constrain final state vars, it changes how we search result vars, so  its by definition seems to me different. And std is not ControlConstraints. So we could create general Constraints with velocities constraints (3 double) and std_constraints (3 double). But how it is now seems readable enough for me. I see SamplingStd - it's clear intention. ControlConstraints - again clear. 
Control constraints is operationally useful with setSpeedLimit controller API, if we make one general constraint struct, we would pass that there too for no reason. For this API we need 3 double struct to make interface clear. 
We can create Constraints .

If I'm not convinced you xD 
We can do something that you propose. But for clear API with setSpeedLimit we need struct that represents that 3 base velocitites and 3 current velocities

Reply:
I don't think that's unreasonable way of thinking, but complimentary, I think having 2 files with just a 3 `double` struct is silly and I think they have natural grouping as max velocity in XYW and the distributions in XYW :laughing: Perhaps a different word than 'constraint' could be used that would help.

Though, I do think the std's are constraints. Not control constraints, but sampling constraints. 

Maybe even just having a single `constraints.hpp` file with the 2 structs: `control constraints` and `sampling constraints`?

Reply:
> Though, I do think the std's are constraints. Not control constraints, but sampling constraints.

I just try to stick to more academic terms xD I've never heard that std is gaussian distribution constraint (May be i'm just not aware). Constraints for me more about optimization field term. 

> I think having 2 files with just a 3 double struct is silly

I don't like that much too. But choosing between single responsibility principle + modularity and a lot of files, i think first one is better. Anyhow i will place them in constraints file, if it feels more comfortable, and we could continue discuss design in next PR